### PR TITLE
Initial ECS Events to Slack lambda script

### PR DESCRIPTION
**Pull request by scout**
I've made `master` into a protected branch to guard from nefarious deployments. Also, the AWS credentials are managed within a CircleCI [context](https://circleci.com/docs/2.0/contexts/#running-workflows-with-a-restricted-context) which means the job can only run if the requester is a member of the alloy-commons GitHub organization. This can be further restricted to a group in CircleCI but a Circle admin would need to create the group and attach it to the context.

### CODE REVIEW

File: lambdas/ecs-events-to-slack/README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Lambda script for ECS event notifications in Slack
+
+This is an [AWS Lambda](https://aws.amazon.com/lambda/) script that processes
+ECS events and sends them to a Slack channel via a webhook url. The webhook url
+for the Slack channel can we set via a `SLACK_WEBHOOK_URL` environment
```
Comment (phrenchphry11) [inline lambdas/ecs-events-to-slack/README.md]: typo? "can we set" -> "can be set"

File: lambdas/ecs-events-to-slack/notify.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
+import json
+import boto3
+import requests
```
Comment (phrenchphry11) [inline lambdas/ecs-events-to-slack/notify.py]: nit but these should be alphabetized!  (weirdly I'm surprised the pep8 linter you were using didn't yell at that!)

File: lambdas/ecs-events-to-slack/notify.py [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+    env_vars = event_detail["overrides"]["containerOverrides"][0]["environment"]
+    env_vars_list = []
+    for env_var in env_vars:
+        env_var_formatted = "%s=\"%s\""
+        % (env_var["name"], env_var["value"])
+        env_vars_list.append(env_var_formatted)
+    env_vars_formatted = "\n".join(env_vars_list)
+
+    logs_url = "https://%s.console.aws.amazon.com/cloudwatch/home?region=%s#logEventViewer:group=%s;stream=ecs/%s/%s"
```
Comment (phrenchphry11) [inline lambdas/ecs-events-to-slack/notify.py]: Okay sorry these are all nits but I slightly prefer this style of string formatting: https://docs.python.org/3/library/stdtypes.html#str.format
So you could do something like "https://{region}.console.aws.amazon.com/cloudwatch/home?region={region}#logEventViewer:group={container_name};stream=ecs/{container_name}/{task_id}".format(region=region, container_name=container_name, task_id=task_id)

actually I don't remember the actual syntax but I think it's something like that :)

File: lambdas/ecs-events-to-slack/notify.py [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+    event_id = event["id"]
+    event_time = event["time"]
+    parsed_time = parse(event_time)
+    formatted_time = str(parsed_time.replace(microsecond=0))
+
+    event_detail = event["detail"]
+    container_name = event_detail["overrides"]["containerOverrides"][0]["name"]
+    task_arn = event_detail["taskArn"]
+    task_id = task_arn.split("/")[1]
```
Comment (phrenchphry11) [inline lambdas/ecs-events-to-slack/notify.py]: Any chance of this index being out-of-bounds?

File: lambdas/ecs-events-to-slack/notify.py [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+            raise e
+        elif e.response["Error"]["Code"] == "InvalidParameterException":
+            print("The request had invalid params:", e)
+            raise e
+        elif e.response["Error"]["Code"] == "InvalidRequestException":
+            print("The request was invalid due to:", e)
+            raise e
+        elif e.response["Error"]["Code"] == "ResourceNotFoundException":
+            print("The requested secret " + secret_name + " was not found")
+            raise e
```
Comment (phrenchphry11) [inline lambdas/ecs-events-to-slack/notify.py]: Soooo I think if you get any other type of ClientError error code, it'll just return nothing - do you want to raise an exception in a generic else statement?

File: lambdas/ecs-events-to-slack/notify.py (hunk 4 above)
Comment (scout) [reply lambdas/ecs-events-to-slack/notify.py]: It _shouldn't_ be out of bounds unless AWS decides to change the `arn` structure...

**Pull request merged**